clicommand/agent_start.go

```diff
@@ -938,6 +946,20 @@ var AgentStartCommand = cli.Command{
 			l.Info("Agents will disconnect after %d seconds of inactivity", agentConf.DisconnectAfterIdleTimeout)
 		}
 
+		// Show allowed environment variables if it is set
+		if len(agentConf.AllowedEnvironmentVariables) > 0 {
+			l.Info("Environment variable allow list is enabled. Only allow-listed environment variables can be set.")
+			l.Debug("Allowed environment variables:")
+
+			for _, v := range agentConf.AllowedEnvironmentVariables {
+				l.Debug(" - %s", v)
+
+				if v[0] != '^' || v[len(v)-1] != '$' {
+					l.Warn(" - %s is not anchored! It can result in environment variable leak with partial match.", v)
+				}
```

Comment:
Thanks for the suggestion! I added the regex validation step to the agent start-up process.

And with the warning about unanchored regex, I agree that it can be okay if the user knows what they are doing but wants to warn as it can be a mistake, and it's always good to have some warnings if it's related to security. 😁 
But it also can get the point as it can be... too much, spamming the terminal.

Also, maybe my wording must not be clear with the message, `environment variable leak with partial match`. I will try to come up with something more explicit. What I meant there is that unanchored regex can end up allowing everything, like `PATH` can end up allowing `LD_LIBRARY_PATH` and any other that includes `PATH` as a partial match is also a regex match. (As it's an allow-list, not deny-list, bad configuration can end up leaking bad env vars to the actual environment.)